python/tk_desktop/desktop_window.py

```diff
@@ -600,6 +605,64 @@ def _register_apps_tab(self):
     def contextMenuEvent(self, event):
         self.user_menu.exec_(event.globalPos())
 
+    def closeEvent(self, event):
+        # When closing the app programatically via self.close(),
+        # the event is not spontaenous and we know the state of self._is_quitting
+        # is correct.
+
+        # When CMD-Q is hit or the close button is pressed however, the dialog
+        # first receives a closeEvent and THEN receives the aboutToQuit signal,
+        # which will call the close event. When this happens, the event is marked
+        # as spontaneous, but we can't tell what actually happened so we postpone
+        # the execution of it by making sure it gets called again after the event
+        # queue is emptied.
```

Comment:
> making sure it gets called again after the event queue is emptied.

I'm not seeing anything here that checks if the event queue is empty. Basically I'm not sure what piece of logic this line is referring to. Is it that the Timer is giving the process enough time to complete all the events?